Assets/MixedRealityToolkit.Tests/EditModeTests/Core/TestFixture_01_MixedRealityToolkitTests.cs

```diff
@@ -594,10 +595,90 @@ public void Test_07_02_DisableServicesByType()
             }
         }
 
+        #region Multiple Instances Tests
+
+        [Test]
+        public void Test_08_01_CreateMultipleInstances()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            MixedRealityToolkit secondInstance = new GameObject("MixedRealityToolkit").AddComponent<MixedRealityToolkit>();
+            MixedRealityToolkit thirdInstance = new GameObject("MixedRealityToolkit").AddComponent<MixedRealityToolkit>();
+
+            Assert.AreNotEqual(secondInstance, MixedRealityToolkit.Instance);
+            Assert.AreNotEqual(thirdInstance, MixedRealityToolkit.Instance);
+            Assert.IsFalse(secondInstance.IsActiveInstance);
+            Assert.IsFalse(thirdInstance.IsActiveInstance);
+        }
+
+        [Test]
+        public void Test_08_02_SwitchBetweenActiveInstances()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            MixedRealityToolkit secondInstance = new GameObject("MixedRealityToolkit").AddComponent<MixedRealityToolkit>();
+            MixedRealityToolkit thirdInstance = new GameObject("MixedRealityToolkit").AddComponent<MixedRealityToolkit>();
+
+            Assert.AreNotEqual(secondInstance, MixedRealityToolkit.Instance);
+            Assert.AreNotEqual(thirdInstance, MixedRealityToolkit.Instance);
+            Assert.IsFalse(secondInstance.IsActiveInstance);
+            Assert.IsFalse(thirdInstance.IsActiveInstance);
+
+            MixedRealityToolkit.SetActiveInstance(secondInstance);
+
+            Assert.AreEqual(secondInstance, MixedRealityToolkit.Instance);
+            Assert.IsTrue(secondInstance.IsActiveInstance);
+
+            MixedRealityToolkit.SetActiveInstance(thirdInstance);
+
+            Assert.AreEqual(thirdInstance, MixedRealityToolkit.Instance);
+            Assert.IsTrue(thirdInstance.IsActiveInstance);
+        }
+
+        [Test]
+        public void Test_08_03_DestroyActiveInstance()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            MixedRealityToolkit secondInstance = new GameObject("MixedRealityToolkit").AddComponent<MixedRealityToolkit>();
+
+            GameObject.DestroyImmediate(MixedRealityToolkit.Instance.gameObject);
+
+            MixedRealityToolkit.SetActiveInstance(secondInstance);
+
+            Assert.NotNull(MixedRealityToolkit.Instance);
+            Assert.AreEqual(secondInstance, MixedRealityToolkit.Instance);
+            Assert.IsTrue(secondInstance.IsActiveInstance);
+        }
+
+        [Test]
+        public void Test_08_04_CreateMultipleInstancesInMultipleScenes()
+        {
+            TestUtilities.CreateScenes(3);
+
+            for (int i = 0; i < SceneManager.sceneCount; i++)
+            {
+                Scene scene = SceneManager.GetSceneAt(i);
+                SceneManager.SetActiveScene(scene);
+
+                TestUtilities.InitializeMixedRealityToolkit();
```

Comment:
Why are you initializing MRTK SceneManager.sceneCount times here?

Reply:
I've created 3 scenes, and for each loaded scene, I'm setting that scene to active and creating an MRTK instance in it. It's to make sure the instances behave as expected across multiple scenes.